testsrc/test262.properties

```diff
@@ -1144,7 +1585,6 @@ built-ins/RegExp 926/1464 (63.25%)
     prototype/Symbol.replace/y-init-lastindex.js
     prototype/Symbol.replace/y-set-lastindex.js
     prototype/Symbol.search/cstm-exec-return-index.js
-    prototype/Symbol.search/failure-return-val.js
```

Comment:
I'm assuming that this is fixed because I didn't check in the file after merging the last PR that fixes a bug in Regexp.

Reply:
I noticed that one too today...

Yeah, if https://github.com/mozilla/rhino/pull/931 fixed something due to which this test now started to pass and that was not accounted for in that PR, if you now rebased your Promises PR on the latest master and regenerated the .properties file, it makes sense that this delta now pops up

Reply:
Just in general (and we can open a new issue for this) -- if someone writes code that breaks test262 tests, I know that the tests will fail the first time, but will the test suite now "fix" test262.properties to indicate that there's a new set of failing tests?

If so, then we will have to monitor every incoming PR to see if it inadvertently disabled stuff in test262.properties.

Reply:
If someone writes code that breaks tests and then they actively regenerate test262.properties (which is a manual thing they have to do, by running the test with the extra `-DupdateTest262properties` argument) and then ignore the delta in test262.properties and put it up as a PR, yeah, then, if if the reviewer also doesn't pay attention, new broken tests might sneak in.

But that is not something new imho: before someone could do something like that manually, by just adding the broken tests to be excluded through test262.properties and put that up for PR.

Reply:
We could do something to prevent stuff like the `prototype/Symbol.search/failure-return-val.js` issue above: we could fail the build if tests excluded in test262.properties aren't failing

Reply:
I most worried now about people introducing new exclusions because they broke things, but the test262 stuff adapter and just added a new exclusion.

What if the default behavior was to remove exclusions from test262.properties automatically, but not add new ones unless you run it with a special system property? That way it's obvious when working that you broke something and it's going to stay broken unless you explicitly did something to add a new exclusions.

Reply:
I guess we could do that. Would we document that 'special' system property or keep that as a sort of undocumented feature then?

Reply:
I think we should document them -- for instance, when adding promises I had to edit the file directly, and that would have had to trigger a "full regen" of the properties file, versus just removing exclusions.

Reply:
To be clear, when we updated the test262 stuff, I somehow thought that the properties file re-gen was going to be optional, but it seems to happen every time I do a "gradle check" which is why I wanted to have some more control.

Reply:
It is optional... Or at least it should be

The regenerate should only happen when the `-DupdateTest262properties` argument is added to `./gradlew tests` and not under any other circumstances

Reply:
Double-checked: the code to update the test262.properties file is only triggered when the `updateTest262properties` System property is set, so not sure how it would get triggered when you run `gradle check`...

Reply:
Indeed, when running the tests through gradle, the regen of test262.properties was always triggered. Added a fix in #951